### ISSUE TITLE
.npmignore

### ISSUE THREAD

**Issue by Tokimon** [compat-table/compat-table#1019]
Hi

I am trying to use this awesome data collection in a small project of mine that detects ES (5, 6, etc.) features used: https://github.com/Tokimon/es-feature-detection

As I fully understand that this doesn't need to be on npm, I am downloading the data directly from Github itself, but that causes me to get everything from the repo, which works but is not ideal. However as I understand it you can define which files npm will download from the github repo by either:

1. Using the .npmignore to exclude all other files than the data files, or
2. Adding a [`files`](https://docs.npmjs.com/files/package.json#files) entry to the package.json to define which files it should include like so
    ```
    {
      ...
      files: [
        'data-common.json'
        'data-es2016plus.js'
        'data-es5.js'
        'data-es6.js'
        'data-esintl.js'
        'data-esnext.js'
        'data-non-standard.js'
      ]
      ...
    }
    ```

I guess downloading the data bases are not entirely what this repo is intended for but it is really a great source for tool like the one I am building to ensure custom builds to be adapted to each project and as such this is just a friendly request. It is not long to implement and would be a nice improvement to an already great repo and I am sure I will not be the only one doing this.

**Comment by ljharb** [comment]
npmignore is the proper way to restrict which files get published to npm (not `files`, which is very dangerous)

That said, using npm to install from github isn't a case we should be worried about supporting, and neither is managing download size of the git repo or npm package.

What files are you suggesting we npmignore?

**Comment by Tokimon** [comment]
hi @ljharb 
Thank you for your reply.
Generally I think people are mostly interested in the `data-` files when they refer to the project. That is my usecase and that is what I have seen elsewhere, which is why I also suggested the `files` method.
So in short, basically everything but the data files should be in the npmignore.

And just to be clear, there is no need to release to npm, but just to add the npmignore.
If you want I can do PR of the npmignore?

Can you elaborate on how `files` is dangerous? I have never really used that feature

**Comment by ljharb** [comment]
`files` is dangerous because the worst case of adding a new file to the repo and forgetting to add it to the whitelist is that the published package will not work, in a way that the author will find hard to detect. `.npmignore`'s worst case is that the package on npm is slightly larger than expected.

Separately, `npm explore foo && npm install && npm test` should always work, for every package.

**Comment by Tokimon** [comment]
Thank you @ljharb for the clarification. It makes sense. Do you want me to do a PR for the `.npmignore`? or do you prefer to create it yourself?

**Comment by ljharb** [comment]
I'm still not convinced it's a good idea to add one. What files, specifically, would you want to include in it?

**Comment by chicoxyzzy** [comment]
I agree that npm module should contain only raw data. One always can `git clone` repo if he or she want HTML files. Size of deps matters for every project. Build scripts and HTML file are useless as part of `node_modules`

**Comment by ljharb** [comment]
The size of deps matters when bundled into the browser - the size of `node_modules` doesn't matter.

Everything is useful if one wants to offline-explore the modules one has already npm-installed.

**Comment by Tokimon** [comment]
Hi @ljharb 
I totally see and understand your point, but as the `compat-table` isn't really on NPM anyway, people who want to checkout you code will clone the project rather than look in the `node_modules` folder.
The `.npmignore` in this case is more for those who wants to use the data in a project and not to explore it.

I agree that in general size doesn't matter when it is in the `node_modules` folder. But it matter a bit in the `npm install` fase, as you don't need to use time on downloading stuff you know you wont need. That is why the `.npmignore` file exists in the first place; to minimize the amount of data/files you have to download, by getting only what is needed to make the module work.

But in the end you are right it doesn't make the big difference if it is there or not, it is more like a nice gesture for the size obsessive programmers like myself (and I guess @chicoxyzzy), that tries to avoid bloat in any sense ;)

**Comment by ljharb** [comment]
If that's the goal, I'd recommend just downloading the exact data files you need instead of trying for the convenience of grabbing the entire repo at once ;-)

**Comment by chicoxyzzy** [comment]
```bash
npm i compat-table
```

\+

```js
var es2016plusData = require('compat-table/data-es2016plus');

createAwesomeAppUsing(es2016plusData);
```
and managing versions via semver and `package.json` is way more easier than downloading exact files, adding them to project's git repo, managing versions by hands.

**Comment by ljharb** [comment]
Yes, it very much it is. And for that reason, the convenience far outweighs the insignificant extra bytes that are downloaded.

**Comment by Tokimon** [comment]
I understand and respect that it is not gonna happen, but for the sake of argument and just to be clear; it is not as much the final weight of the repo as it is the number of files and the idea of only downloading what is needed, so unnecessary calls to the server can be avoided. We already download (more than) enough via `npm i` so each initiative to specify and download only what is needed matters in the grand scale. Which is also why NPM have dependency granularity like `optional dependencies`, `peer dependencies` etc.

Anyway it was just a friendly suggestion to begin with and if you don't want to implement it, it is not that big of a deal. I will still download it as is:

```json
"compat-table": "git+https://github.com/kangax/compat-table.git#gh-pages"
```

**Comment by ljharb** [comment]
(That's not what npm's dep categories are for, other than "deps" vs "devDeps" - optional deps always install but failure doesn't fail the overarching install, and peer deps are required deps that must be installed as singletons at the top level)
